fiftyone/server/routes/sort.py

```diff
@@ -25,31 +24,22 @@ async def post(self, request: Request, data: dict):
         filters = data.get("filters", {})
         stages = data.get("view", None)
         dist_field = data.get("dist_field", None)
-        dataset = fod.load_dataset(dataset_name)
 
-        changed = False
-        if dist_field and not dataset.get_field(dist_field):
-            dataset.add_sample_field(dist_field, fof.FloatField)
-            changed = True
-
-        fosv.get_view(dataset_name, stages=stages, filters=filters)
-
-        state = fose.get_state().copy()
         view = fosv.get_view(dataset_name, stages=stages, filters=filters)
-        state.dataset = view._dataset
+        dataset = view._dataset
 
-        if isinstance(view, fov.DatasetView):
-            state.view = view
-        else:
-            view = None
+        state = fose.get_state().copy()
```

Comment:
I hadn't noticed we create a shallow copy here. Why is it better to do that for sort?

Reply:
I don't recall. Removing it